### PR TITLE
Updating docs for new functional API

### PR DESCRIPTION
**Pull request by scalvert**
[README Rendered](https://github.com/rwjblue/ember-lifeline/blob/7f61730afa41353ccc44e9037fca91794beb9472/README.md)

Additionally cleaned up imports.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,17 +31,43 @@ about the weird parts of working in a long-lived app.
...
+
+  destroy() {
+    runDisposables(this); // ensure that lifeline will clean up any remaining async work
+
+    this._super(...arguments);
+  }
+})
+```
+
+When using the mixin, this is implemented for you.
```
Comment (rwjblue) [inline README.md]: Just skimming this, but I’m not sure if this section is _above_ the mixin docs section. If it is, it’s odd to mention “the mixin” before defining it...

File: README.md [hunk 2]
```diff
@@ -459,7 +534,7 @@ test('updating-time updates', function(assert) {
   return wait()
     .then(() => {
       fakeNow = new Date(2017);
-      pollTaskFor('updating-time#updateTime');
+      pollTaskFor(this._pollToken);
```
Comment (rwjblue) [inline README.md]: FWIW, I still think user provided “labels” are better here 😃

I’m not going to try to convince you, but I do ask that we ensure both are documented....

File: README.md [hunk 3]
```diff
@@ -521,17 +596,17 @@ export default Component.extend(DisposableMixin, {
 });
 ```
 
-This not only adds verbosity to code, but also requires that you symetrically tear down any bindings you setup. By utilizing the `registerDisposable` API, `ember-lifeline` will ensure your registered disposable function will run when the object is destroyed.
+This not only adds verbosity to code, but also requires that you symetrically tear down any bindings you setup. By utilizing the `registerDisposable` API, `ember-lifeline` will ensure your registered disposable function will run when the object is destroyed, provided you call `runDisposables` in your `destroy` method.
```
Comment (rwjblue) [inline README.md]: Maybe:

> provided that you call `runDisposables` during your objects destruction

File: README.md [hunk 4]
```diff
@@ -677,6 +733,40 @@ lifecycle, for example when temporarily dealing with high volume events like `sc
...
+To use the helper using the new ember-qunit module syntax:
+
+```js
+// test-helper.js
+import { module, test } from 'qunit';
+import { setupTest } from 'ember-qunit';
+import setupLifelineValidation from 'ember-lifeline/test-support';
+
+module('module', function(hooks) {
+  setupLifelineValidation(hooks);
```
Comment (rwjblue) [inline README.md]: This example is perfect, but we should emphasize a bit more that `setupLifelineValidation` should be done _before_ `setupTest` / `setupRenderingTesf` etc

File: README.md (hunk 2 above)
Comment (scalvert) [reply README.md]: Bleh. Concede...

**Comment by scalvert** [comment]
@rwjblue ready when you are!

**Pull request merged**